### DISCUSSION TITLE
Error code when calling “ansible-lint *.yml”, for a .yml that is not a playbook structure

### DISCUSSION THREAD

**Discussion by awcrosby** [ansible/ansible-lint#1258] [Ideas]
Currently if a user tries to lint a .yml file, ansible-lint assumes it is a playbook and then outputs any linter issues. If a the .yml file is actually a tasks file such as `tasks/main.yml`, ansible-lint outputs no issues and the user likely thinks the file was linted succesfully. Discussed in #488.

Proposal when calling `ansible-lint *.yml` to check if the format is indeed a playbook format versus tasks file format, and exit with an error code if not a playbook.

**Comment by varac** [comment]
I think https://github.com/varac/ansible-lint-testrepo is a good example of this behaviour.

**Comment by ssbarnea** [comment]
I converted this to a discussion because it is problematic and requires further discussions. Main issue is that it breaks predictive behavior, that any passed file-name is a playbooks. If we implement it as described, it will allow false-positives with yaml files that are no longer playbooks.
